src/compiler/checker.ts

```diff
@@ -13467,32 +13467,37 @@ namespace ts {
 
         function getContextualTypeForJsxExpression(node: JsxExpression): Type {
             // JSX expression can appear in two position : JSX Element's children or JSX attribute
-            const jsxAttributes = isJsxAttributeLike(node.parent) ?
+            const jsxAttributes: JsxAttributes = isJsxAttributeLike(node.parent) ?
                 node.parent.parent :
-                node.parent.openingElement.attributes; // node.parent is JsxElement
+                    isJsxElement(node.parent) ?
+                        node.parent.openingElement.attributes :
+                        undefined; // node.parent is JsxFragment with no attributes
 
-            // When we trying to resolve JsxOpeningLikeElement as a stateless function element, we will already give its attributes a contextual type
-            // which is a type of the parameter of the signature we are trying out.
-            // If there is no contextual type (e.g. we are trying to resolve stateful component), get attributes type from resolving element's tagName
-            const attributesType = getContextualType(jsxAttributes);
+            if (jsxAttributes) {

```

Comment:
Just do an early return of `undefined` here if `jsxAttributes` is undefined

Reply:
Changed.